frontend/packages/kubevirt-plugin/src/components/create-vm/dev-console/dev-console-select-template.tsx

```diff
@@ -0,0 +1,322 @@
+import { VirtualizedGrid } from '@console/shared';
+import { StatusBox } from '@console/internal/components/utils';
+import {
+  getTemplateName,
+  getTemplateProviderType,
+  ProvidedType,
+  templateProviders,
+} from '../../../selectors/vm-template/basic';
+import { V1alpha1DataVolume } from '../../../types/vm/disk/V1alpha1DataVolume';
+import { getTemplateSourceStatus } from '../../../statuses/template/template-source-status';
+import { getTemplateOperatingSystems } from '../../../selectors/vm-template/advanced';
+import { TemplateItem } from '../../../types/template';
+import { isTemplateSourceError } from '../../../statuses/template/types';
+import { usePinnedTemplates } from '../../../hooks/use-pinned-templates';
+import { BOOT_SOURCE_COMMUNITY, BOOT_SOURCE_USER, BOOT_SOURCE_REQUIRED } from '../../../constants';
+import { VMTemplateSupport } from '../../vm-templates/vm-template';
+import { TemplateTile, TemplateTileProps } from '../tabs/select-template';
+import './dev-console-select-template.scss';
+import { HelpIconPopover } from '../../form/popover-helper';
+
+const renderTile = (props: TemplateTileProps & { key: string }) => <TemplateTile {...props} />;
+
+type SelectTemplateProps = {
+  selectTemplate: React.Dispatch<TemplateItem>;
+  pods: PodKind[];
+  dataVolumes: V1alpha1DataVolume[];
+  pvcs: PersistentVolumeClaimKind[];
+  templates: TemplateItem[];
+  namespace: string;
+  loaded: boolean;
+  loadError: any;
+};
+
+const TemplateProviderTitle = (
+  <>
+    <strong>Template Provider</strong>{' '}
+    <HelpIconPopover
+      bodyContent={
+        <Stack hasGutter>
+          <StackItem>
+            <strong>Virtual Machine template types</strong>
+          </StackItem>
+          <StackItem>
+            <strong>Supported</strong> templates are Red Hat tested and certified. Customers may
+            contact Red Hat for support.
+          </StackItem>
+        </Stack>
+      }
+    />
+  </>
+);
+
+const BootSourcesTitle = (
+  <>
+    <strong>Boot Sources</strong>{' '}
+    <HelpIconPopover
+      bodyContent={
+        <Stack hasGutter>
+          <StackItem>
+            <strong>Boot source types</strong>
+          </StackItem>
+          <StackItem>
+            A <strong>Red Hat</strong> defined boot source is added to the cluster via the operator.
+          </StackItem>
+          <StackItem>
+            A <strong>Community</strong> defined boot source is added to the cluster via the
+            operator.
+          </StackItem>
+          <StackItem>
+            A <strong>User</strong> defined boot source is added to the cluster by a user.
+          </StackItem>
+        </Stack>
+      }
+    />
+  </>
+);
+
+export const DevConsoleSelectTemplate: React.FC<SelectTemplateProps> = ({
+  selectTemplate,
+  pods,
+  dataVolumes,
+  templates,
+  pvcs,
+  loaded,
+  loadError,
+}) => {
+  const [isPinned] = usePinnedTemplates();
+  const [filters, setFilters] = React.useState<{
```

Comment:
i guess it would be handy to make hook out of these filters & logic around since we use the same in admin side